### DISCUSSION TITLE
From the output of occ it seems it could generate makefiles for me?

### DISCUSSION THREAD

**Discussion by ghost** [LADSoft/OrangeC#796] [General]
```
Dependency generation:
  /M             - basic generation
  /MM            - basic generation, user files only
  /MF file       - specify output file
  /MG            - missing headers as dependencies
  /MP            - add phony targets
  /MT target     - add target
Press <ENTER> to continue...
  /MQ target     - add target, quote special characters
  /MD            - basic generation and continue
  /MMD           - basic generation and continue, user files only
```

How to use this feature? It maybe the last chance I give to `omake`. It crashed with any of the makefiles written by me. There is also no documents for it or a list of differences with gnu make so I'm pretty much in dark corner here. I have no idea why it crashed and don't have any directions to adjust the makefiles to make it works. Maybe it would work with the makefiles generated by `occ`.

**Comment by GitMensch** [comment]
The author of OrangeC already said that he will have a look into the omake issues tonight... and actually this is a very quick response time.
So I'd suggest to wait for omake until this is done, which _possibly_ even is "your tomorrow".

Those switches will generate dependency makefiles for you, these are commonly included in a bigger Makefile with the result that if any header used changed (o)make can directly compile every source that include it.
Note: this feature is likely identical to the gcc implementation, so if you want to know more about that you can also have a look there, for example:

https://programmer.group/gcc-m-mm-mmd-mf-mt.html
https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html
